api/envoy/extensions/filters/http/compressor/v3/compressor.proto

```diff
@@ -132,8 +132,16 @@ message Compressor {
 }
 
 // Per-route overrides of `ResponseDirectionConfig`. Anything added here should be optional,
-// to allow overriding arbitrary subsets of configuration. Omitted fields must have no affect.
+// to allow overriding arbitrary subsets of configuration. Omitted fields must have no effect.
 message ResponseDirectionOverrides {
+  // If true, removes accept-encoding from the request headers before dispatching it to the upstream
+  // so that responses do not get compressed before reaching the filter.
+  //
+  // .. attention::
+  //
+  //    To avoid interfering with other compression filters in the same chain use this option in
+  //    the filter closest to the upstream.
+  bool remove_accept_encoding_header = 3;
```

Comment:
Please add the `optional` keyword to this field definition.

The intent of `ResponseDirectionOverrides` is to make it possible to override only what one wants to override, and in particular to make it possible to enable compressor filter per-route, without otherwise changing anything.

There's a reasonable config that would be broken by this change:
  1. You have the compressor disabled at the top level, with `remove_accept_encoding_header: true`.
  2. You have a per-route override providing an empty `response_direction_config`, which enables the compressor.
  3. Now, with this field newly added, the `remove_accept_encoding_header` setting will be overridden.

The reason you don't see `optional` fields is that the proto3 language lacked that feature when Envoy started, so Envoy had to adopt a more cumbersome and less efficient approach when optional scalar fields were required. #14973 discussed adopting `optional`. Idk what further discussion has happened.

If the maintainers are unable to approve the use of `optional`, then we'll need to use the obsolete WKT wrappers.

Reply:
Cool.  I will go ahead and start using `optional`.  We'll see what everyone thinks.

Reply:
Thanks for the review, by the way.

Reply:
Welp.  In `//test/tools/type_whisperer:api_type_db_test`:
```
envoy/extensions/filters/http/compressor/v3/compressor.proto: is a proto3 file that contains optional fields, but code generator protoc-gen-api_proto_plugin hasn't been updated to support optional fields in proto3. Please ask the owner of this code generator to support proto3 optional.
```

Reply:
This seems like a question for @envoyproxy/api-shepherds  @htuch  AFAIK we haven't moved forward from https://github.com/envoyproxy/envoy/issues/14973

Reply:
I would like us to move to optional scalar, but someone needs to take #14973 and run with it. I think more than the open discussion there, there is legit work to be done to answer the open questions and fix tooling like our plugins. So, help wanted. For this review, I would just stick with WKT as elsewhere in the API.

Reply:
Can anyone give me an example of achieving optionals with WKT in the codebase?  Blindly grepping didn't get me anything when I tried a while ago.

Reply:
I'm just suggesting `google.protobuf.BoolValue ...`

Reply:
Thanks.  I know nothing about protobuf or C++.

Reply:
This is now using WKT.